### PR TITLE
[MM-50453] Fixes on E2E tests

### PR DESCRIPTION
**Pull request by fmartingr**
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-50453

#### Output

<details>
<summary>Output</summary>

```
=== RUN   TestMattermostImport
    e2e_test.go:190: validate the environment and gather variables
    e2e_test.go:204: Upload the archive for translation
time="2023-02-14T11:05:20+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:05:21+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:05:22+01:00" level=info msg="Waiting for upload to complete.."
    e2e_test.go:218: clean up the environment from any previously interrupted tests
    e2e_test.go:240: failed with status code 400
    e2e_test.go:240: failed with status code 400
    e2e_test.go:246: create an Installation into which to run an import
    e2e_test.go:274: wait for the Installation to become stable
    e2e_test.go:316: start a new translation into installation g57i44afqpbm5derdyo3q645oy
    e2e_test.go:325: create a new translation with key hg5i94pq378ujguh3496enxxww.zip
    e2e_test.go:337: wait for translation rje9m1eskbd43jmad17twj7djc to start
    e2e_test.go:371: wait for translation rje9m1eskbd43jmad17twj7djc to complete
    e2e_test.go:426: make sure an import is created and wait for it to start
    e2e_test.go:392: wait for import to move from import-requested state to installation-pre-adjustment
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-pre-adjustment state to import-in-progress
    e2e_test.go:406: wait for installation to be in import-in-progress state
    e2e_test.go:406: wait for installation to be in import-complete state
    e2e_test.go:392: wait for import to move from import-in-progress state to import-complete
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:461: Checking import status
    e2e_test.go:392: wait for import to move from import-complete state to installation-post-adjustment
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-post-adjustment state to import-succeeded
    e2e_test.go:528: check teams
    e2e_test.go:560: check channels
    e2e_test.go:659: check users
    e2e_test.go:612: check posts
--- PASS: TestMattermostImport (534.79s)
PASS
ok      github.com/mattermost/awat/test/e2e     535.182s
go test -v -tags e2e -count 1 ./test/e2e -timeout 30m -run Slack
=== RUN   TestSlackTranslationAndImport
    e2e_test.go:190: validate the environment and gather variables
    e2e_test.go:204: Upload the archive for translation
time="2023-02-14T11:14:17+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:14:18+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:14:19+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:14:20+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:14:21+01:00" level=info msg="Waiting for upload to complete.."
    e2e_test.go:218: clean up the environment from any previously interrupted tests
    e2e_test.go:246: create an Installation into which to run an import
    e2e_test.go:274: wait for the Installation to become stable
    e2e_test.go:316: start a new translation into installation mzupzq6csifmuygpkj7ypxpair
    e2e_test.go:325: create a new translation with key 3rur7863aid6fk77tsxfzfhz8e.zip
    e2e_test.go:337: wait for translation i91pbbonubyyjf8fubiga3be4a to start
    e2e_test.go:371: wait for translation i91pbbonubyyjf8fubiga3be4a to complete
    e2e_test.go:426: make sure an import is created and wait for it to start
    e2e_test.go:392: wait for import to move from import-requested state to installation-pre-adjustment
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-pre-adjustment state to import-in-progress
    e2e_test.go:406: wait for installation to be in import-in-progress state
    e2e_test.go:406: wait for installation to be in import-complete state
    e2e_test.go:392: wait for import to move from import-in-progress state to import-complete
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:461: Checking import status
    e2e_test.go:392: wait for import to move from import-complete state to installation-post-adjustment
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-post-adjustment state to import-succeeded
    e2e_test.go:528: check teams
    e2e_test.go:560: check channels
    e2e_test.go:659: check users
    e2e_test.go:612: check posts
--- PASS: TestSlackTranslationAndImport (537.41s)
PASS
ok      github.com/mattermost/awat/test/e2e     537.843s
go test -v -tags e2e -count 1 ./test/e2e -timeout 30m -run TwoInQuickSuccession
=== RUN   TestTwoInQuickSuccession
    e2e_test.go:190: validate the environment and gather variables
    e2e_test.go:204: Upload the archive for translation
time="2023-02-14T11:23:17+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:23:18+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:23:19+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:23:20+01:00" level=info msg="Waiting for upload to complete.."
time="2023-02-14T11:23:21+01:00" level=info msg="Waiting for upload to complete.."
    e2e_test.go:218: clean up the environment from any previously interrupted tests
    e2e_test.go:246: create an Installation into which to run an import
    e2e_test.go:274: wait for the Installation to become stable
    e2e_test.go:246: create an Installation into which to run an import
    e2e_test.go:274: wait for the Installation to become stable
    e2e_test.go:316: start a new translation into installation khd74odu33y4m8ootjuri5wq8c
    e2e_test.go:325: create a new translation with key ptdpfttbfjfcpcfsfg4o1xad7c.zip
    e2e_test.go:316: start a new translation into installation xotmtse4cp8bbmhu5m6x4j9r7c
    e2e_test.go:325: create a new translation with key ptdpfttbfjfcpcfsfg4o1xad7c.zip
    e2e_test.go:337: wait for translation nxfxigw5kffytmizb86re33wfr to start
    e2e_test.go:337: wait for translation mg37imdc5t8eibgk481p4tcu1h to start
    e2e_test.go:371: wait for translation mg37imdc5t8eibgk481p4tcu1h to complete
    e2e_test.go:371: wait for translation nxfxigw5kffytmizb86re33wfr to complete
    e2e_test.go:426: make sure an import is created and wait for it to start
    e2e_test.go:426: make sure an import is created and wait for it to start
    e2e_test.go:392: wait for import to move from import-requested state to installation-pre-adjustment
    e2e_test.go:392: wait for import to move from import-requested state to installation-pre-adjustment
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-pre-adjustment state to import-in-progress
    e2e_test.go:406: wait for installation to be in import-in-progress state
    e2e_test.go:406: wait for installation to be in import-complete state
    e2e_test.go:392: wait for import to move from import-in-progress state to import-complete
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:461: Checking import status
    e2e_test.go:392: wait for import to move from import-complete state to installation-post-adjustment
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-pre-adjustment state to import-in-progress
    e2e_test.go:406: wait for installation to be in import-in-progress state
    e2e_test.go:406: wait for installation to be in import-complete state
    e2e_test.go:392: wait for import to move from import-in-progress state to import-complete
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:461: Checking import status
    e2e_test.go:392: wait for import to move from import-complete state to installation-post-adjustment
    e2e_test.go:392: wait for import to move from installation-post-adjustment state to import-succeeded
    e2e_test.go:406: wait for installation to be in stable state
    e2e_test.go:392: wait for import to move from installation-post-adjustment state to import-succeeded
    e2e_test.go:101: validate the import data was imported properly
    e2e_test.go:104: checking import into installation xotmtse4cp8bbmhu5m6x4j9r7c
    e2e_test.go:528: check teams
    e2e_test.go:560: check channels
    e2e_test.go:659: check users
    e2e_test.go:612: check posts
    e2e_test.go:104: checking import into installation khd74odu33y4m8ootjuri5wq8c
    e2e_test.go:528: check teams
    e2e_test.go:560: check channels
    e2e_test.go:659: check users
    e2e_test.go:612: check posts
--- PASS: TestTwoInQuickSuccession (872.59s)
PASS
ok      github.com/mattermost/awat/test/e2e     873.020s
````

</details>

#### Release Note

```release-note
None
```

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -703,12 +702,22 @@ func checkUsers(env *completedEnvironment) {
 }
 
 // if the doer returns true, consider it done, and stop retrying
-func retryFor(d time.Duration, doer func() bool) {
```
Comment (fmartingr) [inline test/e2e/e2e_test.go:706]: This was changed to a timer/ticker approach, which seems more golang-friendly, and it now fails directly if the timeout is met, which wasn't previously unless the desired state was checked after this method call (which wasn't in some cases).

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -427,9 +428,6 @@ func waitForImportToSucceed(
 	importStatusList, err := client.GetImportStatusesByTranslation(ts.ID)
 	require.Equal(t, 1, len(importStatusList))
 	importStatus := importStatusList[0]
-	if importStatus.StartAt == 0 {
```
Comment (fmartingr) [inline test/e2e/e2e_test.go:430]: This is already checked below in "step 1", unsure why this is here.

**Review by mirshahriar — APPROVED** [review]

File: test/e2e/e2e_test.go (hunk 2 above)
Comment (mirshahriar) [reply test/e2e/e2e_test.go:430]: We don't need this.

**Review by mirshahriar — APPROVED** [review]

**Review by gabrieljackson — APPROVED** [review]
Very nice. LGTM

**Pull request merged**